### PR TITLE
[ARG] Support enabling ARG for clients

### PR DESCRIPTION
**Pull request by lzhecheng**
* Add enabledARG field in client config to support ARG
* Update related doc and template

Signed-off-by: Zhecheng Li <zhechengli@microsoft.com>

#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
* Add enabledARG field in client config to support ARG
* Update related doc and template

related: #1493
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
Support enabling ARG for clients
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/azureclients/azure_client_config.go [hunk 1]
```diff
@@ -61,6 +63,15 @@ type RateLimitConfig struct {
 	CloudProviderRateLimitBucketWrite int `json:"cloudProviderRateLimitBucketWrite,omitempty" yaml:"cloudProviderRateLimitBucketWrite,omitempty"`
 }
 
+func (cfg *ClientConfig) CheckARG(enabledARGClients map[string]bool, client string) *ClientConfig {
+	newClientConfig := *cfg
+	if enabled, ok := enabledARGClients[client]; ok && enabled {
+		klog.V(2).Infof("ARG is enabled for %s", client)
```
Comment (andyzhangx) [inline pkg/azureclients/azure_client_config.go:69]: lower log level to v(5)

Comment (andyzhangx) [reply pkg/azureclients/azure_client_config.go:69]: also, it's possible that `enabledARGClients` is nil, then panic?

Comment (lzhecheng) [reply pkg/azureclients/azure_client_config.go]: log level updated.
`enabledARGClients` can be nil but I don't think it will panic.

Comment (andyzhangx) [reply pkg/azureclients/azure_client_config.go]: would you add a ut where `enabledARGClients` is nil? Thanks.

Comment (lzhecheng) [reply pkg/azureclients/azure_client_config.go]: Updated.

**Comment by lzhecheng** [comment]
@andyzhangx could you please take another look?

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Comment by lzhecheng** [comment]
The design has changed and this PR is not needed. Closing.

**Pull request closed without merging**